spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java

```diff
@@ -235,31 +238,43 @@ public void before() throws Exception { //NOSONAR
 				this.kafkaPorts[i] = TestUtils.boundPort(server, SecurityProtocol.PLAINTEXT);
 			}
 		}
-		addTopics(this.topics);
+		createKafkaTopics(this.topics, (topics) -> {
+		});
 		System.setProperty(SPRING_EMBEDDED_KAFKA_BROKERS, getBrokersAsString());
 		System.setProperty(SPRING_EMBEDDED_ZOOKEEPER_CONNECT, getZookeeperConnectionString());
 	}
 
 	/**
-	 * Add topics to the existing broker(s) using the configured number of partitions.
+	 * Create topics in the existing broker(s) using the configured number of partitions.
 	 * @param topics the topics.
-	 * @since 2.1
+	 * @param callback the callback.
 	 */
-	public void addTopics(String... topics) {
+	private void createKafkaTopics(Set<String> topics, java.util.function.Consumer<Set<String>> callback) {
 		doWithAdmin(admin -> {
-			List<NewTopic> newTopics = Arrays.stream(topics)
+			List<NewTopic> newTopics = topics.stream()
 					.map(t -> new NewTopic(t, this.partitionsPerTopic, (short) this.count))
 					.collect(Collectors.toList());
 			CreateTopicsResult createTopics = admin.createTopics(newTopics);
 			try {
 				createTopics.all().get();
+				callback.accept(topics);
 			}
 			catch (Exception e) {
 				throw new KafkaException(e);
 			}
 		});
 	}
 
+
+	/**
+	 * Add topics to the existing broker(s) using the configured number of partitions.
+	 * @param topics the topics.
+	 * @since 2.1
+	 */
+	public void addTopics(String... topics) {
+		createKafkaTopics(new HashSet<>(Arrays.asList(topics)), this.topics::addAll);
```

Comment:
I don't think we need such a `callback` overhead. You simply can call this `addAll()` in the end of this `addTopics()` method. More over in other place the callback is just empty, so, it really doesn't bring too much value in the logic of this class.